### PR TITLE
Replace APIv7 calls

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix MANAGER-8564, MANAGER-8566, MANAGER-8577, MANAGER-8579, MANAGER-8567, MANAGER-8568, MANAGER-8572, MANAGER-5631
| License          | BSD 3-Clause

## Description

Replace APIv7 calls by APIv6 calls

### CODE REVIEW

**Review by Jisay — APPROVED** [review]

**Review by frenautvh — APPROVED** [review]

**Review by JeremyDec — DISMISSED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/phone/codec/codec.controller.js [hunk 1]
```diff
@@ -186,15 +185,21 @@ export default /* @ngInject */ function TelecomTelephonyLinePhoneCodecCtrl(
...
+              billingAccount: service.billingAccount,
+              serviceName: service.serviceName,
+            })
+            .$promise.then(() => service)
+            .catch(() => null);
         }),
-      ),
-    );
+      )
+      .then((results) => compact(results));
```
Comment (zakariasahmane) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/phone/codec/codec.controller.js:202]: Is compact necessary? If you want to remove null values u could just do : 

```
.then((results) => results.filter((result) => result))
```

It's better than importing another thing from lodash again

File: packages/manager/apps/telecom/src/app/telecom/pack/slots/exchangeAccount/exchangeAccount.controller.js [hunk 2]
```diff
@@ -1,66 +1,93 @@
...
 
-  init();
+  getExchangeAccounts() {
+    return this.getExchangeAccountServices()
+      .then((exchangeServicesIds) => {
+        const accountIdsByExchangeService = exchangeServicesIds.reduce(
+          (all, { exchangeService, accountId }) => {
+            return {
+              ...all,
+              [exchangeService]: [...(all[exchangeService] || []), accountId],
```
Comment (zakariasahmane) [inline packages/manager/apps/telecom/src/app/telecom/pack/slots/exchangeAccount/exchangeAccount.controller.js:56]: This line is a bit hard to understand, a comment would me nice to improve maintainability

**Comment by JeremyDec** [comment]
Hello @cbourgois, there is failure on test phase (linked to Semver task).

**Comment by cbourgois** [comment]
> Hello @cbourgois, there is failure on test phase (linked to Semver task).

Should be OK now.

**Review by JeremyDec — DISMISSED** [review]

**Comment by cbourgois** [comment]
Rebased on `master`

**Review by marie-j — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Pull request merged**